src/include/sof/audio/audio_stream.h

```diff
@@ -46,15 +46,23 @@ struct sof_audio_stream_params {
 	uint32_t rate;		/**< Number of data frames per second [Hz] */
 	uint16_t channels;	/**< Number of samples in each frame */
 
-	/** alignment limit of stream copy, this value indicates how many
-	 * integer frames which can meet both byte align and frame align
-	 * requirement. it should be set in component prepare or param functions
+	/**
+	 * Alignment limit of stream copy, used by audio_stream_avail_frames_aligned.
+	 * We can set byte_align and frame_align in the init or prepare functions to
+	 * limit the processing data size is byte_align bytes aligned and frame_align
+	 * frames aligned. This feature indicates how many integer frames can meet both
+	 * byte_align and frame_align requirement.
 	 */
 	uint16_t frame_align;
 
 	/**
-	 * alignment limit of stream copy, alignment is the frame_align_shift-th
-	 * power of 2 bytes. it should be set in component prepare or param functions
+	 * Alignment limit of stream copy, used by audio_stream_avail_frames_aligned.
+	 * We use right shifting frame_align_shift bits to replace divising framesize to
+	 * avoid runtime division. That means, if the frame size and the frame_align
+	 * we set both are power of 2, we will get the same results by shifting and
+	 * division. Otherwise we will get smaller available frame number than division,
+	 * and processing data size is the frame_align_shift-th power of 2 bytes aligned.
+	 * It should be set in component prepare or param functions
```

Comment:
same, you are making the explanations more verbose without explaining the concepts.

Reply:
@andrula-song adding some example context is also helpful to explain. e.g. 
```
....using xtensa HiFi3 then its' best we align frame sizes of X bytes, e.g. a 16 bit stereo stream with Hifi3 would be Y frames and Z bytes....using Hifi4 then ......
```